### PR TITLE
devcontainer: init

### PR DESCRIPTION
**Pull request by mjibson**
Create devcontainer scaffolding and document how to use.

### Motivation

  * This PR adds a feature that has not yet been specified.

### Testing

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

File: misc/python/materialize/cli/mzcompose.py [hunk 1]
```diff
@@ -404,6 +404,7 @@ def run(self, args: argparse.Namespace) -> None:
             )
             .strip()
             .strip("v")
+            .split("+")[0]
```
Comment (mjibson) [inline misc/python/materialize/cli/mzcompose.py:407]: `2.6.0+azure-1` is the version string produced that this fixes.

File: misc/scratch/remote-dev.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "vs code remote dev",
+  "launch_script": "true",
+  "instance_type": "c6a.4xlarge",
```
Comment (mjibson) [inline misc/scratch/remote-dev.json:4]: This is the cheapest amd64 16 vcpu box on ec2. 8 cores was just not enough when I was using the default dev-box as a daily driver.

File: misc/scratch/remote-dev.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  "ami": "ami-09d56f8956ab235b3",
+  "size_gb": 100,
```
Comment (mjibson) [inline misc/scratch/remote-dev.json:6]: I also quickly ran out of disk with only 50g, so doubling for regular remote dev.

File: doc/developer/scratch.md [hunk 4]
```diff
@@ -83,9 +83,24 @@ If you need to, you can install and use the `mssh` command provided by the
...
+4. From your laptop, ensure ssh works and the fingerprint is correct: `ssh ubuntu@INSTANCE-IP-ADDRESS`
+5. Open VS Code on your laptop, and open `Remote-SSH: Connect to Host...` from the command palatte.
+6. Click to add a new ssh host and enter `ssh ubuntu@INSTANCE-IP-ADDRESS` and add it to your local ssh config.
+7. Click `Connect` in the `Host added!` dialog.
+8. Open the `materialize` folder (which is automatically created by the scratch command), then `Reopen in Container` from the dialog.
+
+This will take a few minutes to build your dev container.
+See the [guide](https://code.visualstudio.com/docs/remote/ssh) for more details.
+Because this is in scratch, it may be deleted after a week.
+If you need a longer term enviroment, figure out how to do that and edit this line.
```
Comment (benesch) [inline doc/developer/scratch.md:103]: You can pass a different `--max-age-days` value when you create your scratch instance.

**Review by benesch — APPROVED** [review]

File: doc/developer/scratch.md (hunk 4 above)
Comment (mjibson) [reply doc/developer/scratch.md]: :+1:

**Pull request merged**